### DISCUSSION TITLE
Ability to require a record

### DISCUSSION THREAD

**Discussion by JamesChevalier** [excid3/noticed#372] [Ideas]
We can `require_params :param_name` but `require_params :record` isn't functional - even if you set that `record` value it still produces the validation error
```
Param `record` is required for CommentNotifier. (Noticed::ValidationError)
```

I _think_ this is because the `params` are being adjusted here, removing that `record` entry from the overall `params` hash:
https://github.com/excid3/noticed/blob/f8e0f8cb0aba9cf6284042bd504e16eb70845067/app/models/concerns/noticed/deliverable.rb#L48-L51
So by the time the required params check runs, the record has been taken out & the validation will fail.

**Comment by excid3** [comment]
That's correct. I think we may add an exception for `:record` so that it checks the database attributes instead of params instead.

**Reply by JamesChevalier** [reply]
Sounds good.
We should also update the readme to remove :record from the `required_params` entry in [this section](https://github.com/excid3/noticed/blob/main/README.md#required-params) if the syntax is going to be different.

**Reply by excid3** [reply]
I think we'll leave it with `required_params :record`.

**Reply by excid3** [reply]
Actually... now that this is just an ActiveRecord model, maybe this should just become:

```ruby
validates :record, presence: true
```

Then there's nothing fancy at all going on.

**Reply by excid3** [reply]
That works perfectly. Updated the readme!

**Comment by fig** [comment]
I think the docs could more explicitly point out that `require_params :record` is **not** the correct thing to do and will error.
I just got caught out by this and was pulling my hair out until I found this discussion.
I'm happy to submit a PR if you would like.
